site/src/components/WorkspaceDeletion/DeletionText.tsx

```diff
@@ -0,0 +1,34 @@
+import { Workspace } from "api/typesGenerated"
+import { displayImpendingDeletion } from "./utils"
+import { useDashboard } from "components/Dashboard/DashboardProvider"
+import styled from "@emotion/styled"
+import { Theme as MaterialUITheme } from "@mui/material/styles"
+
+export const DeletionText = ({
+  workspace,
+}: {
+  workspace: Workspace
+}): JSX.Element | null => {
+  const { entitlements, experiments } = useDashboard()
+  const allowAdvancedScheduling =
+    entitlements.features["advanced_template_scheduling"].enabled
+  // This check can be removed when https://github.com/coder/coder/milestone/19
+  // is merged up
+  const allowWorkspaceActions = experiments.includes("workspace_actions")
+
+  if (
+    !displayImpendingDeletion(
+      workspace,
+      allowAdvancedScheduling,
+      allowWorkspaceActions,
+    )
+  ) {
+    return null
+  }
+  return <StyledSpan role="status">Impending deletion</StyledSpan>
+}
+
+const StyledSpan = styled.span<{ theme?: MaterialUITheme }>`
+  color: ${(props) => props.theme.palette.warning.light};
+  font-weight: 600;
```

Comment:
In MUI v5 you can style things like ChakraUI:

```tsx
<Box
  component="span"
  fontWeight={600}
  color={theme => theme.palette.warning.light}
>
  Impending Deletion
</Box>
```

or 

```tsx
<Box
  component="span"
  sx={{
    fontWeight: 600,
    color: theme.palette.warning.light,
  }}
>
  Impending Deletion
</Box>
```

Just to let you know in case you find it better